File: dotnet/Allmark/Types/InlineRule.cs
namespace Allmark.Types;

/// <summary>
/// Defines a rule for parsing inline markdown elements.
/// </summary>
public record InlineRule
{
	public required string Name { get; init; }
	public required Func<InlineParserState, MarkdownNode, bool> Test { get; init; }
}

File: zig/src/types/InlineRule.zig
const std = @import("std");

pub const InlineRule = struct {
    const Self = @This();

    const InlineParserState = @import("InlineParserState.zig").InlineParserState;
    const MarkdownNode = @import("MarkdownNode.zig").MarkdownNode;

    name: []const u8,

    @"test": *const fn (state: *InlineParserState, parent: *MarkdownNode) bool,
};
